### ISSUE TITLE
Type naming

### ISSUE THREAD

**Issue by sethvincent** [digidem/mapeo-schema#38]
A couple of things to decide related to type naming:

- should we include the schema version in the type name?
- should we include the encoding in the type name? json vs protobuf

Some notes I had on this from the #32 pull request:

It might be ideal to have a naming convention like:

- `ObservationJson`
- `ObservationProtobuf`

Related: it looks like the protobuf comments export names like `CoreOwnership_1` rather than `CoreOwnership`.

I have mixed feelings on whether the public types should include the version in the name. The main place that might matter is in migration code where it's needed to be explicit about versions and use multiple versions in the same code.

Maybe we could import latest version as just the name:

```js
/**
* @property {import('mapeo-schema/types').ObservationJson} Observation
*/

/**
* @property {import('mapeo-schema/types').ObservationProtobuf} Observation
*/
```

As well as being more specific about the version needed:

```js
/**
* @property {import('mapeo-schema/types').ObservationJson4} Observation
*/

/**
* @property {import('mapeo-schema/types').ObservationJson5} Observation
*/
```

A type without a version could be the latest version.

I can imagine there's a good argument for always including the version number and not having the convenience of importing the latest version by just the schema name.

**Comment by tomasciccola** [comment]
protobuf names should be unique for a project, so that's the reason they have the version on the name. I'll probably rename the export so that the last version is the default one with the possibility of importing a specific version

**Comment by sethvincent** [comment]
closed by #43
